packages/ai-chat/src/browser/chat-tool-preferences.ts

```diff
@@ -0,0 +1,178 @@
+import {
+    createPreferenceProxy,
+    PreferenceProxy,
+    PreferenceService,
+    PreferenceSchema,
+    PreferenceContribution
+} from '@theia/core/lib/browser/preferences';
+
+/**
+ * Enum for tool confirmation modes
+ */
+export enum ToolConfirmationMode {
+    YOLO = 'yolo',
+    CONFIRM = 'confirm',
+    DISABLED = 'disabled'
+}
+
+export const TOOL_CONFIRMATION_PREFERENCE = 'ai-features.chat.toolConfirmation';
+
+export const chatToolPreferences: PreferenceSchema = {
+    type: 'object',
+    properties: {
+        [TOOL_CONFIRMATION_PREFERENCE]: {
+            type: 'object',
+            additionalProperties: {
+                type: 'string',
+                enum: [ToolConfirmationMode.YOLO, ToolConfirmationMode.CONFIRM, ToolConfirmationMode.DISABLED],
+                enumDescriptions: [
+                    nls.localize('theia/ai/chat/toolConfirmation/yolo/description', 'Execute tools automatically without confirmation'),
+                    nls.localize('theia/ai/chat/toolConfirmation/confirm/description', 'Ask for confirmation before executing tools'),
+                    nls.localize('theia/ai/chat/toolConfirmation/disabled/description', 'Disable tool execution')
+                ]
+            },
+            default: {},
+            description: nls.localize('theia/ai/chat/toolConfirmation/description',
+                'Configure confirmation behavior for different tools. Key is the tool ID, value is the confirmation mode.' +
+                'Use "*" as the key to set a global default for all tools.'),
+            title: AI_CORE_PREFERENCES_TITLE,
+        }
+    }
+};
+
+export interface ChatToolConfiguration {
+    [TOOL_CONFIRMATION_PREFERENCE]: { [toolId: string]: ToolConfirmationMode };
+}
+
+export const ChatToolPreferenceContribution = Symbol('ChatToolPreferenceContribution');
+export const ChatToolPreferences = Symbol('ChatToolPreferences');
+export type ChatToolPreferences = PreferenceProxy<ChatToolConfiguration>;
+
+export function createChatToolPreferences(preferences: PreferenceService, schema: PreferenceSchema = chatToolPreferences): ChatToolPreferences {
+    return createPreferenceProxy(preferences, schema);
+}
+
+export function bindChatToolPreferences(bind: interfaces.Bind): void {
+    bind(ChatToolPreferences).toDynamicValue((ctx: interfaces.Context) => {
+        const preferences = ctx.container.get<PreferenceService>(PreferenceService);
+        const contribution = ctx.container.get<PreferenceContribution>(ChatToolPreferenceContribution);
+        return createChatToolPreferences(preferences, contribution.schema);
+    }).inSingletonScope();
+    bind(ChatToolPreferenceContribution).toConstantValue({ schema: chatToolPreferences });
+    bind(PreferenceContribution).toService(ChatToolPreferenceContribution);
+}
+
+/**
+ * Utility class to manage tool confirmation settings
+ */
+@injectable()
+export class ToolConfirmationManager {
```

Comment:
nit: moving this into its own file

Reply:
Will be done in a follow-up -> Move to Core